tests/abc_atlas_cache/test_cache.py

```diff
@@ -0,0 +1,573 @@
+
+        expected_path = self.cache_dir / 'data/data_file.txt'
+
+        url = f'http://{self.test_bucket_name}.s3.amazonaws.com/data/data_file.txt'  # noqa: E501
+        good_attributes = CacheFileAttributes(url=url,
+                                              version=self.new_version,
+                                              file_size=1234,
+                                              local_path=expected_path,
+                                              relative_path=relative_path,
+                                              file_type='txt',
+                                              file_hash=true_checksum)
+
+        assert not expected_path.exists()
+        cache._download_file(good_attributes)
+        assert expected_path.exists()
+        hasher = hashlib.md5()
+        with open(expected_path, 'rb') as in_file:
+            hasher.update(in_file.read())
+        assert hasher.hexdigest() == true_checksum
+
+    def test_re_download_file(self):
+        """
+        Test that S3CloudCache._download_file will re-download a file
+        when it has been removed from the local system
+        """
+        data = b'11235813kjlssergwesvsdd'
+        true_checksum = hash_data(data)
+        relative_path = 'data/data_file.txt'
+
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=relative_path,
+                               Body=data)
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        expected_path = self.cache_dir / relative_path
+
+        url = f'http://{self.test_bucket_name}.s3.amazonaws.com/data/data_file.txt'  # noqa: E501
+        good_attributes = CacheFileAttributes(url=url,
+                                              version=self.new_version,
+                                              file_size=1234,
+                                              local_path=expected_path,
+                                              relative_path=relative_path,
+                                              file_type='txt',
+                                              file_hash=true_checksum)
+
+        assert not expected_path.exists()
+        cache._download_file(good_attributes)
+        assert expected_path.exists()
+        hasher = hashlib.md5()
+        with open(expected_path, 'rb') as in_file:
+            hasher.update(in_file.read())
+        assert hasher.hexdigest() == true_checksum
+
+        # now, remove the file, and see if it gets re-downloaded
+        expected_path.unlink()
+        assert not expected_path.exists()
+
+        cache._download_file(good_attributes)
+        assert expected_path.exists()
+        hasher = hashlib.md5()
+        with open(expected_path, 'rb') as in_file:
+            hasher.update(in_file.read())
+        assert hasher.hexdigest() == true_checksum
+
+    def test_download_data(self):
+        """
+        Test that S3CloudCache.download_data() correctly downloads files from
```

Comment:
This test no longer (or never did) downloads the data. It just checks that the `CacheFileAttributes` is correctly formulated (specifically, it asserts that the files it inquires about do not exist). Can you update the docstring?